adapter/internal/oasparser/envoyconf/routes_with_clusters.go

```diff
@@ -625,6 +625,16 @@ func processEndpoints(clusterName string, clusterDetails *model.EndpointCluster,
 		cluster.MaxRequestsPerConnection = wrapperspb.UInt32(1)
 	}
 
+	// If the endpoint is within the cluster, set the set max connection duration for given env variable
+	if withinClusterEndpoint && os.Getenv("ROUTER_MAX_CONNECTION_DURATION_SECONDS") != "" {
+		maxConnectionDuration, err := strconv.Atoi(os.Getenv("ROUTER_MAX_CONNECTION_DURATION_SECONDS"))
+		if err != nil {
+			logger.LoggerOasparser.Error("Error while converting max connection duration to int. ", err)
+		} else {
+			cluster.CommonHttpProtocolOptions.MaxConnectionDuration = durationpb.New(time.Duration(maxConnectionDuration) * time.Second)
```

Comment:
As per the last discussion we had, once the max duration is hit, no new requests will be processed through the connection, but the existing request will be served completely (eg: when there's a high response time)
What is the value we are planning to put?
To make sure, shall we also test with an API which is having about 3 mins response time, keep sending requests for sometime and make sure it works without an issue?

Reply:
> As per the last discussion we had, once the max duration is hit, no new requests will be processed through the connection, but the existing request will be served completely (eg: when there's a high response time)

Yes correct

> What is the value we are planning to put?

I am initially planning to test out with 5 seconds and see the throughput difference.

> To make sure, shall we also test with an API which is having about 3 mins response time, keep sending requests for sometime and make sure it works without an issue?

Yes that was the plan, set the env var to 5 seconds, and have a workload with prestop sleep of 8 seconds and see how it perform under the load.

Reply:
> Yes that was the plan, set the env var to 5 seconds, and have a workload with prestop sleep of 8 seconds and see how it perform under the load.

Ack, I was also referring to testing an API with a long response time as well (without prestop/redeploying etc). This is just to make sure long running API requests are not impacted by this config.